### PR TITLE
Fix centos test cases

### PR DESCRIPTION
**Pull request by shweta83**
It fixes C2R test cases.
Another related PR for master is failing in stream and 6.14 which will take some time to merge so I created another PR for 6.13.z and below releases which should be fixed.

### CODE REVIEW

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k test_convert2rhel_centos

**Comment by shweta83** [comment]
Closed by mistake

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k test_convert2rhel_centos

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k test_convert2rhel_centos

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k test_convert2rhel_centos

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k test_convert2rhel_centos

File: tests/foreman/api/test_convert2rhel.py [hunk 1]
```diff
@@ -267,7 +278,7 @@ def test_convert2rhel_oracle(target_sat, oracle, activation_key_rhel, version):
 
 
 @pytest.mark.e2e
-@pytest.mark.parametrize("version", ['centos7', 'centos8'], indirect=True)
+@pytest.mark.parametrize("version", ['centos7'], indirect=True)
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_convert2rhel.py:281]: why removing centos8? and do we not test centos9 for C2R yet?
```suggestion
@pytest.mark.parametrize('version', ['centos7'], indirect=True)
```

File: tests/foreman/api/test_convert2rhel.py [hunk 2]
```diff
@@ -287,8 +298,8 @@ def test_convert2rhel_centos(target_sat, centos, activation_key_rhel, version):
     """
     host_content = target_sat.api.Host(id=centos.hostname).read_json()
     major = version.split('.')[0]
-    assert host_content['operatingsystem_name'] == f"CentOS {major}"
 
+    assert host_content['operatingsystem_name'] == f"CentOS {major}"
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_convert2rhel.py:302]:
```suggestion
    assert host_content['operatingsystem_name'] == f'CentOS {major}'
```

File: tests/foreman/api/test_convert2rhel.py [hunk 3]
```diff
@@ -70,6 +70,7 @@ def register_host(sat, act_key, module_org, module_loc, host, ubi=None):
         location=module_loc,
         insecure=True,
         repo=ubi,
+        force=True,
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_convert2rhel.py:73]: why force register?

File: tests/foreman/api/test_convert2rhel.py [hunk 4]
```diff
@@ -155,6 +162,10 @@ def centos(
     ]
     act_key = create_activation_key(target_sat, module_org, module_lce, cv, c2r_sub.id)
     register_host(target_sat, act_key, module_org, smart_proxy_location, centos_host)
+    # Below lines are workaround because of the kernel mismatch errror.
+    # These will be removed once we have the solution  to fix this.
+    centos_host.execute("yum update kernel -y")
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_convert2rhel.py:167]:
```suggestion
    centos_host.execute("yum update kernel* -y")
```

**Review by Gauravtalreja1 — COMMENTED** [review]
@shweta83 Added few questions/suggestions, and PR looks duplicate of your old PR https://github.com/SatelliteQE/robottelo/pull/10840 so please close either of them, also @ogajduse has few comments added there so please address them too.

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k test_convert2rhel_centos

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k test_convert2rhel_centos

**Comment by shweta83** [comment]
All the comments are addressed in https://github.com/SatelliteQE/robottelo/pull/10840/files and closing this PR.

**Pull request closed without merging**